### PR TITLE
[imporve][doc] Update cluster failover document

### LINKED ISSUE #17433: [Doc] Cluster-level failover shouldn't be part of Java client chapter
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### What issue do you find in Pulsar docs?

When I read the Java client document, I found a very long chapter to introduce [cluster-level failover](https://pulsar.apache.org/docs/next/client-libraries-java#cluster-level-failover). Seems that cluster-level failover is a feature of Pulsar rather than the Java client, so I think this part needs to be moved to another place.

### What is your suggestion?

I think we should move the introduction of `cluster-level failover` to the next chapter of [Geo Replication](https://pulsar.apache.org/docs/next/concepts-replication), and just give code snippets in client library pages (if supported).

cc @tisonkun @Anonymitaet @momo-jun @codelipenghui 

### Any reference?

https://pulsar.apache.org/docs/next/client-libraries-java#cluster-level-failover

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by labuladong**
Fixes #17433 

### Modification

Create a new chapter "Cluster-level failover" behind the Geo-Replication chapter and only keep code demos in Java doc.

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `doc-not-needed` 
(Please explain why)
  
- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Comment by labuladong** [comment]
cc @momo-jun @Anonymitaet

**Comment by tisonkun** [comment]
@labuladong if this patch is ready for review, you can remove the "WIP:" prefix. And I suggest that you use Draft PR mechanism to state status.

File: site2/docs/client-libraries-java.md [hunk 1]
```diff
@@ -1654,3 +1341,140 @@ PulsarClient client = PulsarClient.builder()
     .build();
 ```
 
+## Cluster-level failover
+
+More introduction for cluster-level failover please look at [Cluster-level failover](concepts-cluster-level-failover.md) chapter.
```
Comment (momo-jun) [inline site2/docs/client-libraries-java.md:1346]:
```suggestion
For more concepts and reference information about cluster-level failover, see [Cluster-level failover](concepts-cluster-level-failover.md). 
```

File: site2/docs/concepts-cluster-level-failover.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+![Workflow of controlled failover cluster](/assets/cluster-level-failover-5.png)
+
+</TabItem>
+
+</Tabs>
+````
+
+### How to use cluster-level failover
+
+Only Java client  support cluster-level failover so far, please look at [Java client document](client-libraries-java.md#cluster-level-failover).
```
Comment (momo-jun) [inline site2/docs/concepts-cluster-level-failover.md:163]:
```suggestion
Only Java clients support cluster-level failover at this moment. See [Java client document](client-libraries-java.md#cluster-level-failover).
```

File: site2/docs/client-libraries-java.md (hunk 1 above)
Comment (momo-jun) [reply site2/docs/client-libraries-java.md:1346]: @labuladong can you pls attach the screenshots for these updates in the PR description?

Comment (Anonymitaet) [inline site2/docs/client-libraries-java.md]:
```suggestion
For more information about cluster-level failover, including concept, benefits, use cases, constraints, usage, and working principles, see [Cluster-level failover](concepts-cluster-level-failover.md). 
```

**Review by Anonymitaet — COMMENTED** [review]
Thanks for your contribution!

Before submitting a doc PR, please [preview your changes locally](https://docs.google.com/document/d/1wszdtMRo6MhKbVaggPK7_bnKaC4TewuT--GWZZxJNGg/edit#). When submitting a PR, please attach screenshots of your local preview. In this way, you can get your PR reviewed and merged more quickly.

Thank you! 😊

**Comment by labuladong** [comment]
@momo-jun @Anonymitaet  Thanks for your review! Screenshots:

image


image

image

**Review by Anonymitaet — APPROVED** [review]

**Comment by labuladong** [comment]
All tests passed. Could you help to merge this pr? @Anonymitaet

**Review by momo-jun — APPROVED** [review]

**Pull request merged**